### DISCUSSION TITLE
Piplines not work

### DISCUSSION THREAD

**Discussion by ByteAndBits** [Security-Onion-Solutions/securityonion#11259] [2.4]
### Version

2.4.100

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

64

### RAM

256GB

### Storage for /

Raid5 

### Storage for /nsm

Raid10

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

Hi,

we installed Sec Onion in a distributed setup. We want grook some syslogs. Pattern works fine in Grook debugger.
Syslog is send to standard syslog port, elesatic entry shows piple: syslog.

So we add our grock pattern to this default pipline. But it dosn´t work. No pharsing form the message field.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by ByteAndBits** [comment]
Edit: In "in all in one" Server test  it worked. Now in distributed setup it doesn´t. 1x mange Server, 1x receive node, 1x forwarder noder, 2x search node.

Where we can get support? This is a advanced test, if we don´t get it work we change to other tools for our data center.
We spend a lot of time, but found no answer in documentation or somewhere.

We don´t pay money for enterprise support without possitive test results. 
wazuh worked, solarwinds worked, Darktrace worked, Security Onioin not in all points

**Comment by dougburks** [comment]
> we installed Sec Onion in a distributed setup. We want grook some syslogs. Pattern works fine in Grook debugger.
Syslog is send to standard syslog port, elesatic entry shows piple: syslog.
So we add our grock pattern to this default pipline. But it dosn´t work. No pharsing form the message field.

Before we get too far into grok, what kind of logs are we talking about? Have you checked to see if there is already an existing Elastic integration that will do the parsing for you?
https://docs.securityonion.net/en/2.4/syslog.html
https://docs.securityonion.net/en/2.4/elastic-agent.html#elastic-agent
https://docs.securityonion.net/en/2.4/elastic-fleet.html#elastic-fleet
https://docs.elastic.co/integrations

**Comment by ByteAndBits** [comment]
It is standard Syslog from sophos utm. Syslog message are shown in elastic, with metafileds. The intergration of UTM don´t work at all. 
We try it, but no messages shown in Elastic. UTM is experimental for integration. So we switched back to syslog.

For us it´s strange. In SecOnion 2.3 standalone worked, 2.4 (beta) standalone it worked, 2.4 distributed not. 
This feature is for us importent. We also catch a lot of syslog from ot devices with no integration. 

Is there a point where we can start the debugging?
![syslog](https://github.com/Security-Onion-Solutions/securityonion/assets/118199753/5691fee6-6380-4a90-b850-b8c81936443a)

**Reply by jwmasekre** [reply]
To confirm, when you say you wrote to the pipeline, do you mean the one at `/opt/so/conf/elasticsearch/ingest/syslog`?

**Comment by ByteAndBits** [comment]
Yes, I just added a line in the default syslog grok processor 


![sys](https://github.com/Security-Onion-Solutions/securityonion/assets/118199753/dffe9c3a-c18c-4da0-a191-2e0a0327491b)

**Comment by ByteAndBits** [comment]
Ok with elastic integration works basic (Firewall settings are missing at the right point). For us are the wrong informations will be parsed. So we need to do it with syslog and grok or a other way to parse out the information we need. Same problem we have with our switches. So how we can go on here?

**Comment by ByteAndBits** [comment]
Found it!

A big mess on our side. We have big network and syslog traffic at the same time (2.5M syslog entries in 8 Days). So we get double entries for the syslog messages. One form forwarder node an on from firewall. forwarder node was faster than syslog. so we "saw" only the network traffic syslog message in live dicovery. 
A special filter from a colleague helped to see the right entries. And it worked as expected.
